src/server/session.ts

```diff
@@ -870,20 +868,37 @@ namespace ts.server {
         }
 
         /** It is the caller's responsibility to verify that `!this.suppressDiagnosticEvents`. */
-        private updateErrorCheck(next: NextStep, checkList: PendingErrorCheck[], ms: number, requireOpen = true) {
+        private updateErrorCheck(next: NextStep, checkList: readonly string[] | readonly PendingErrorCheck[], ms: number, requireOpen = true) {

```

Comment:
if `createCheckList` is deleted, is `updateErrorCheck` *ever* called with anything but `string[]`? If so, maybe the other callers should stop creating the check list ahead of time too?

Reply:
The only reason it can be checklist ahead of time is when scheduling errors for a project.. Which is when we get file names in a project and create checklist that says get errors for all files in project if and only if that file is in project..